File: backend-springboot/src/main/java/com/javeriana/pontimovil/ponti_movil/dto/gestion_rutas/ruta_enviada/rRutaEnviadaDTO.java
package com.javeriana.pontimovil.ponti_movil.dto.gestion_rutas.ruta_enviada;

import com.javeriana.pontimovil.ponti_movil.dto.gestion_rutas.ruta_recibida.rHorarioDTO;
import lombok.*;
import java.io.Serializable;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@EqualsAndHashCode
@NoArgsConstructor
public class rRutaEnviadaDTO implements Serializable {
    UUID id;
    String codigo;
    rHorarioDTO horario;
    List<String> diasSemana;
    List<rEstacionEnviadaDTO> estaciones;

    @Override
    public String toString() {
        return "rRutaEnviadaDTO{" +
                "id=" + id +
                ", codigo='" + codigo + '\'' +
                ", horario=" + horario +
                ", diasSemana=" + diasSemana +
                ", estaciones=" + estaciones +
                '}';
    }
}

File: frontend-angular/src/app/dto/gestionar-rutas/ruta-enviada/r-ruta-enviada-dto.ts
import {RHorarioDto} from '../ruta-recibida/r-horario-dto';
import {REstacionRecibidaDto} from '../ruta-recibida/r-estacion-recibida-dto';

export class RRutaEnviadaDto {
  constructor(
    public id: string,
    public codigo: string,
    public horario: RHorarioDto,
    public diasSemana: string[],
    public estaciones: REstacionRecibidaDto[]
  ) {}
}
